api/src/services/github/github.service.ts

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+import { Octokit } from 'octokit';
+
+import config from 'config';
+
+const urlParams = {
+  gitHub:'https://github.com/login/oauth',
+  client_id: config.github.clientId,
+  redirect_uri: `${config.apiUrl}/account/sign-in/github`,
+  client_secret: config.github.clientSecret,
+};
+
+const oAuthURL = `${urlParams.gitHub}/authorize?client_id=${urlParams.client_id}&redirect_uri=${urlParams.redirect_uri}&scope=user`;
+
+const exchangeCodeForToken = async (code: string) => {
+  try {
+    const { data } = await axios.request({
+      url: `${urlParams.gitHub}/access_token?client_id=${urlParams.client_id}&client_secret=${urlParams.client_secret}&code=${code}&redirect_uri=${urlParams.redirect_uri}&scope=user`,
+      method: 'post',
+      headers: {
+        accept: 'application/json',
+        'content-type': 'application/json',
+      },
+    });
+
+    const octokit = new Octokit({
+      auth: data.access_token,
+    });
+
+    const { data : { name, email } }  = await octokit.request('GET /user', {});
+    let primaryEmail = email;
+
+    if (!email) {
+      const emails = await octokit.request('GET /user/emails', {});
+
+      primaryEmail = emails.data.find((em)=> em.primary)?.email || '';
```

Comment:
Can we have primary email or any available email instead of empty string?

Reply:
Yes, i'll fix it!